filament/src/RenderTarget.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "details/RenderTarget.h"
+
+#include "details/Engine.h"
+#include "details/Texture.h"
+
+#include "FilamentAPI-impl.h"
+
+#include <utils/Panic.h>
+
+namespace filament {
+
+using namespace details;
+
+struct RenderTarget::BuilderDetails {
+    FRenderTarget::Attachment mAttachments[AttachmentPoint::COUNT];
+};
+
+using BuilderType = RenderTarget::Builder;
+BuilderType::Builder() noexcept = default;
+BuilderType::~Builder() noexcept = default;
+BuilderType::Builder(Builder const& rhs) noexcept = default;
+BuilderType::Builder(Builder&& rhs) noexcept = default;
+BuilderType& BuilderType::operator=(BuilderType const& rhs) noexcept = default;
+BuilderType& BuilderType::operator=(BuilderType&& rhs) noexcept = default;
+
+BuilderType& BuilderType::texture(AttachmentPoint pt, Texture* texture) noexcept {
+    mImpl->mAttachments[pt].texture = upcast(texture);
+    return *this;
+}
+
+BuilderType& BuilderType::mipLevel(AttachmentPoint pt, uint8_t level) noexcept {
+    mImpl->mAttachments[pt].mipLevel = level;
```

Comment:
Technically we need to validate pt here because AttachmentPoint::COUNT is "valid" at the call site.

Reply:
Good point, to fix this I moved it out of the enum.